### PR TITLE
CCO-367: Add periodic for AWS STS

### PR DESCRIPTION
**Pull request by suhanime**
Add a weekly periodic to test STS and alert failures or errors via Slack.

xref: https://issues.redhat.com/browse/CCO-367

/assign @2uasimojo

### CODE REVIEW

File: ci-operator/jobs/openshift/cloud-credential-operator/openshift-cloud-credential-operator-master-periodics.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    ci.openshift.io/generator: prowgen
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-openshift-cloud-credential-operator-master-e2e-aws-manual-oidc-cron
+  reporter_config:
+    slack:
+      channel: '#team-hive-alert'
+      job_states_to_report:
+      - failure
+      - error
+      report_template: Job {{ .Spec.Job }} failed. {{ .Status.URL|View logs }}
```
Comment (suhanime) [inline ci-operator/jobs/openshift/cloud-credential-operator/openshift-cloud-credential-operator-master-periodics.yaml:22]: @2uasimojo I want to tag hive-cop here but I either don't know how to escape `@` in yaml or `make update` doesn't like it. Any suggestions?

**Comment by suhanime** [comment]
Will need to add the periodic to 4.13+ if the tests pass

**Comment by suhanime** [comment]
/pj-rehearse periodic-ci-openshift-cloud-credential-operator-master-e2e-aws-manual-oidc-cron

**Comment by 2uasimojo** [comment]
This looks correct. Internally discussing whether the signal:noise ratio makes this inviable though.

**Comment by 2uasimojo** [comment]
@abutcher is helping try to reduce flakes/failures via https://github.com/openshift/origin/pull/27887. If that brings the success rate up to something workable, we'll merge this. Otherwise we'll keep looking at shoring things up first.

/hold

**Comment by sdodson** [comment]
/retitle CCO-367: Add periodic for AWS STS

**Comment by 2uasimojo** [comment]
Gonna run all of these again. Unclear at this point whether https://github.com/openshift/origin/pull/27895 is going to be necessary; and would like to see if the 4.11 and 4.12 ones are just flakes.

/pj-rehearse periodic-ci-openshift-cloud-credential-operator-release-4.10-e2e-aws-manual-oidc-cron
/pj-rehearse periodic-ci-openshift-cloud-credential-operator-release-4.11-e2e-aws-manual-oidc-cron
/pj-rehearse periodic-ci-openshift-cloud-credential-operator-release-4.12-e2e-aws-manual-oidc-cron
/pj-rehearse periodic-ci-openshift-cloud-credential-operator-release-4.13-e2e-aws-manual-oidc-cron
/pj-rehearse periodic-ci-openshift-cloud-credential-operator-release-4.14-e2e-aws-manual-oidc-cron
/pj-rehearse periodic-ci-openshift-cloud-credential-operator-release-4.15-e2e-aws-manual-oidc-cron

**Comment by sdodson** [comment]
4.11 and 4.12 failed on the upgradable condition being off, not sure why 4.13 didn't fail there as I'd expect since the test wasn't updated to tolerate this.

```
{  fail [github.com/onsi/ginkgo/v2@v2.1.5-0.20220909190140-b488ab12695a/internal/suite.go:612]: Apr 25 22:54:32.057: Some cluster operators are not ready: cloud-credential (Upgradeable=False MissingUpgradeableAnnotation: Upgradeable annotation cloudcredential.openshift.io/upgradeable-to on cloudcredential.operator.openshift.io/cluster object needs updating before upgrade. See Manually Creating IAM documentation for instructions on preparing a cluster for upgrade.)
Ginkgo exit error 1: exit with code 1}
```

**Comment by suhanime** [comment]
@abutcher Need your eyes and guidance on this one

**Comment by abutcher** [comment]
We backported https://github.com/openshift/origin/pull/27887 to 4.13 but for 4.12, 4.11 to allow CCO to be `upgradeable=false` expect we will need to backport to those branches as well.

**Comment by suhanime** [comment]
/pj-rehearse periodic-ci-openshift-cloud-credential-operator-release-4.12-e2e-aws-manual-oidc-cron

**Comment by suhanime** [comment]
/pj-rehearse periodic-ci-openshift-cloud-credential-operator-release-4.10-e2e-aws-manual-oidc-cron

**Comment by suhanime** [comment]
/pj-rehearse periodic-ci-openshift-cloud-credential-operator-release-4.11-e2e-aws-manual-oidc-cron

**Comment by suhanime** [comment]
@abutcher @2uasimojo This is ready to go in

**Pull request merged**